### PR TITLE
[SMALLFIX]  some refactoring

### PR DESCRIPTION
**Pull request by peisun1115**

### CODE REVIEW

File: core/server/src/main/java/alluxio/worker/block/BlockWorkerClientServiceHandler.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public Void call() throws AlluxioException, IOException {
    */
   @Override
   public LockBlockResult lockBlock(final long blockId, final long sessionId)
-      throws AlluxioTException {
+     throws AlluxioTException {
```
Comment (ifcharming) [inline core/server/src/main/java/alluxio/worker/block/BlockWorkerClientServiceHandler.java]: indent?

Comment (peisun1115) [inline core/server/src/main/java/alluxio/worker/block/BlockWorkerClientServiceHandler.java]: fixed

**Comment by peisun1115** [comment]
Will fix the test. Hold on review. I am marking it as WIP

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
@calvinjia @ifcharming 

This PR is ready again.  Thank you.

File: core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java [hunk 2]
```diff
@@ -71,7 +79,7 @@ public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long
           .writeAndFlush(new RPCBlockReadRequest(blockId, offset, length, lockId, sessionId));
       channelFuture = channelFuture.sync();
       if (channelFuture.isDone() && !channelFuture.isSuccess()) {
-        LOG.error("Failed to write to %s for block %d with error %s.", address.toString(), blockId,
+        LOG.error("Failed to write to %s for block %d with error %s.", mAddress.toString(), blockId,
```
Comment (apc999) [inline core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java]: is this `LOG` message correctly using `%s` rather than `{}`? Feel free to address this later to keep this  PR clean

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by ifcharming** [comment]
LGTM

File: core/client/src/main/java/alluxio/client/RemoteBlockReader.java [hunk 3]
```diff
@@ -40,13 +35,7 @@ private Factory() {} // prevent instantiation
      * @return a new instance of {@link RemoteBlockReader}
      */
     public static RemoteBlockReader create() {
-      try {
-        return CommonUtils.createNewClassInstance(
-            Configuration.<RemoteBlockReader>getClass(PropertyKey.USER_BLOCK_REMOTE_READER_CLASS),
```
Comment (apc999) [inline core/client/src/main/java/alluxio/client/RemoteBlockReader.java]: do we depreciate `USER_BLOCK_REMOTE_READER_CLASS`? If yes, could you comment in `PropertyKey.java` for this property?

File: core/client/src/main/java/alluxio/client/RemoteBlockWriter.java [hunk 4]
```diff
@@ -39,13 +35,7 @@ private Factory() {} // prevent instantiation
      * @return a new instance of {@link RemoteBlockWriter}
      */
     public static RemoteBlockWriter create() {
-      try {
-        return CommonUtils.createNewClassInstance(
-            Configuration.<RemoteBlockWriter>getClass(PropertyKey.USER_BLOCK_REMOTE_WRITER_CLASS),
```
Comment (apc999) [inline core/client/src/main/java/alluxio/client/RemoteBlockWriter.java]: Same comment too for deprecation

File: core/client/src/main/java/alluxio/client/block/RemoteBlockOutStream.java [hunk 5]
```diff
@@ -53,8 +58,8 @@ public RemoteBlockOutStream(long blockId,
     super(blockId, blockSize, blockStoreContext);
     mCloser = Closer.create();
     try {
-      mRemoteWriter = mCloser.register(RemoteBlockWriter.Factory.create());
       mBlockWorkerClient = mCloser.register(mContext.createWorkerClient(address));
+      mRemoteWriter = mCloser.register(RemoteBlockWriter.Factory.create());
```
Comment (apc999) [inline core/client/src/main/java/alluxio/client/block/RemoteBlockOutStream.java]: why change the order here? is it by purpose?
if there is something tricky, we should put a comment here

File: core/common/src/main/java/alluxio/AbstractThriftClient.java [hunk 6]
```diff
@@ -165,9 +169,10 @@
    *
    * @param client the client
    * @param e the exception
+   * @throws E if the exception is not suppressed
    */
-  protected void processException(C client, Exception e) {
-    // Empty implementation by default.
+  protected <E extends Exception> void processException(C client, E e) throws E {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/AbstractThriftClient.java]: do we really need this method? it is only used in this abstract class but never being extended

Comment (apc999) [inline core/common/src/main/java/alluxio/AbstractThriftClient.java]: do we really need this method? it is only used in this abstract class but never being extended

File: tests/src/test/java/alluxio/client/FileInStreamIntegrationTest.java [hunk 7]
```diff
@@ -57,9 +58,12 @@
   @BeforeClass
   public static final void beforeClass() throws Exception {
     sFileSystem = sLocalAlluxioClusterResource.get().getClient();
-    sWriteBoth = CreateFileOptions.defaults().setWriteType(WriteType.CACHE_THROUGH);
-    sWriteAlluxio = CreateFileOptions.defaults().setWriteType(WriteType.MUST_CACHE);
-    sWriteUnderStore = CreateFileOptions.defaults().setWriteType(WriteType.THROUGH);
+    sWriteBoth = CreateFileOptions.defaults().setMode(Mode.createFullAccess())
```
Comment (apc999) [inline tests/src/test/java/alluxio/client/FileInStreamIntegrationTest.java]: is this change required? how it works before this PR as I don't see real code change related in this PR?

Comment (peisun1115) [inline tests/src/test/java/alluxio/client/FileInStreamIntegrationTest.java]: ack

File: core/common/src/main/java/alluxio/AbstractThriftClient.java (hunk 6 above)
Comment (peisun1115) [inline core/common/src/main/java/alluxio/AbstractThriftClient.java]: ack

File: core/client/src/main/java/alluxio/client/block/RemoteBlockOutStream.java (hunk 5 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/block/RemoteBlockOutStream.java]: ack

File: core/client/src/main/java/alluxio/client/RemoteBlockReader.java (hunk 3 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/RemoteBlockReader.java]: i just deleted them

File: core/client/src/main/java/alluxio/client/RemoteBlockWriter.java (hunk 4 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/RemoteBlockWriter.java]: done
